packages/angular_devkit/schematics/tools/file-system-utility.ts

```diff
@@ -6,48 +6,27 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {
-  FileDoesNotExistException,
-  JsonParseMode,
-  JsonValue,
-  parseJson,
-} from '@angular-devkit/core';
+import { JsonValue } from '@angular-devkit/core';
+import { FileDoesNotExistException } from '@angular-devkit/schematics';
 import { existsSync, readFileSync } from 'fs';
+import { ParseError, parse, printParseErrorCode } from 'jsonc-parser';
 
-/**
- * Read a file and returns its content. This supports different file encoding.
- */
-export function readFile(fileName: string): string {
-  if (!existsSync(fileName)) {
-    throw new FileDoesNotExistException(fileName);
+export function readJsonFile(path: string): JsonValue {
+  if (!existsSync(path)) {
+    throw new FileDoesNotExistException(path);
   }
-  const buffer = readFileSync(fileName);
-  let len = buffer.length;
-  if (len >= 2 && buffer[0] === 0xfe && buffer[1] === 0xff) {
-    // Big endian UTF-16 byte order mark detected. Since big endian is not supported by node.js,
-    // flip all byte pairs and treat as little endian.
-    len &= ~1;
-    for (let i = 0; i < len; i += 2) {
-      const temp = buffer[i];
-      buffer[i] = buffer[i + 1];
-      buffer[i + 1] = temp;
-    }
 
-    return buffer.toString('utf16le', 2);
-  }
-  if (len >= 2 && buffer[0] === 0xff && buffer[1] === 0xfe) {
-    // Little endian UTF-16 byte order mark detected
-    return buffer.toString('utf16le', 2);
-  }
-  if (len >= 3 && buffer[0] === 0xef && buffer[1] === 0xbb && buffer[2] === 0xbf) {
-    // UTF-8 byte order mark detected
-    return buffer.toString('utf8', 3);
-  }
+  const errors: ParseError[] = [];
+  const content = parse(readFileSync(path, 'utf-8'), errors, { allowTrailingComma: true });
```

Comment:
Does `jsonc-parser` automatically handle the UTF-8 byte order mark?

Reply:
Yes it does.

Reply:
![Screenshot 2021-08-12 at 19 50 21](https://user-images.githubusercontent.com/17563226/129244681-b7575159-ecf2-4a81-9991-50bcdeabf6a4.png)